### PR TITLE
Validate number of args in command

### LINKED ISSUE #1628: analytics command accepts an argument but expects a flag
**Describe the bug**
The analytics command accepts an argument (like disable) but expects only flags.

**To Reproduce**

1. Run `okteto analytics disable`
2. See ` ✓  Analytics have been enabled`

**Expected behavior**
I would expect the command to exit with a non-zero status code and a message along the lines of "unexpected argument" along with the help being printed out

**Desktop (please complete the following information):**

```console
➜  ~ okteto version  
okteto version 1.13.0
```

### PR DESCRIPTION
**Pull request by jLopezbarb**
Signed-off-by: Javier López Barba <javier@okteto.com>

Fixes #1628

## Proposed changes
- Validates the number of arguments allowed to enter a command and returns an error with the help of the command.
-

### CODE REVIEW

File: cmd/build.go [hunk 1]
```diff
@@ -38,7 +39,13 @@ func Build(ctx context.Context) *cobra.Command {
 	var secrets []string
 
 	cmd := &cobra.Command{
-		Use:   "build [PATH]",
+		Use: "build [PATH]",
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) >= 2 {
+				return errors.New("okteto build requires one or no argument")
```
Comment (pchico83) [inline cmd/build.go:45]: Can we have the error in `errors.go` to make it easier to review CLI error messages by other teams?

File: cmd/login.go [hunk 2]
```diff
@@ -31,6 +31,7 @@ func Login() *cobra.Command {
 	token := ""
 	cmd := &cobra.Command{
 		Use:   "login [url]",
+		Args:  cobra.MaximumNArgs(1),
```
Comment (pchico83) [inline cmd/login.go:34]: how does the error look like in this case?

File: cmd/build.go (hunk 1 above)
Comment (pchico83) [reply cmd/build.go:45]: could we use `cobra.MaximumNArgs(1)`?

File: cmd/login.go (hunk 2 above)
Comment (jLopezbarb) [reply cmd/login.go:34]: Captura de pantalla 2021-06-30 a las 17 23 14

**Review by pchico83 — APPROVED** [review]

**Comment by rberrelleza** [comment]
That error is not very actionable, is it possible to customize the text per command to give a hint to the user? Specially for commands that don't take arguments.

**Comment by jLopezbarb** [comment]
@rberrelleza @melissa-okteto I'm not satisfied with the hint that we can give to the users. The error messages given are as follows:
- For no args command:
Captura de pantalla 2021-07-01 a las 15 53 14
- For commands with maximum number of args:
Captura de pantalla 2021-07-01 a las 16 03 22
- For command with minimum number of args: 
Captura de pantalla 2021-07-01 a las 16 04 19
- For commands with a specified number of args:
Captura de pantalla 2021-07-01 a las 16 05 29

For no args command without documentation page the hint is hidden which is the part that I don't like because it's not consistent to show the documentation on some parts and not in others, so if you can think of what to show on the hint, any suggestion will be welcome .

**Comment by melissa-okteto** [comment]
@jLopezbarb, when the hint is hidden, is there anything shows up? The messages above seem clear.

> For no args command without documentation page the hint is hidden which is the part that I don't like because it's not consistent to show the documentation on some parts and not in others, so if you can think of what to show on the hint, any suggestion will be welcome .

**Comment by jLopezbarb** [comment]
if the hint is hidden we don't show any text other than the error in red

**Comment by rberrelleza** [comment]
> @rberrelleza @melissa-okteto I'm not satisfied with the hint that we can give to the users. The error messages given are as follows:
> 
> * For no args command:
> 
> Captura de pantalla 2021-07-01 a las 15 53 14
> 
> - For commands with maximum number of args:
> Captura de pantalla 2021-07-01 a las 16 03 22
> 
> - For command with minimum number of args:
> Captura de pantalla 2021-07-01 a las 16 04 19
> 
> - For commands with a specified number of args:
> Captura de pantalla 2021-07-01 a las 16 05 29
> 
> For no args command without documentation page the hint is hidden which is the part that I don't like because it's not consistent to show the documentation on some parts and not in others, so if you can think of what to show on the hint, any suggestion will be welcome .

What if we just show "Invalid command" or "Invalid arguments" and show the help of the command?

**Comment by rberrelleza** [comment]
Could we just apply this to ` okteto analytics`? I feel like for all the other commands the error we already display when you put the number of arguments is good enough.

**Comment by pchico83** [comment]
> Could we just apply this to ` okteto analytics`? I feel like for all the other commands the error we already display when you put the number of arguments is good enough.

We are not validating arguments for most commands. For example, you can do `okteto push xxx` and it doesn't fail

**Pull request merged**